infra/main.bicep

```diff
@@ -0,0 +1,140 @@
+// "resourceGroupName": {
+//      "value": "myGroupName"
+// }
+param appName string = ''
+param applicationInsightsDashboardName string = ''
+param applicationInsightsName string = ''
+param appServicePlanName string = ''
+param mysqlServerName string = ''
+param mysqlAdminName string = 'petclinic'
+@secure()
+param mysqlAdminPassword string
+param mysqlDatabaseName string = 'petclinic'
+param keyVaultName string = ''
+param logAnalyticsName string = ''
+param resourceGroupName string = ''
+
+@description('Id of the user or app to assign application roles')
+param principalId string = ''
+
+var abbrs = loadJsonContent('./abbreviations.json')
+var resourceToken = toLower(uniqueString(subscription().id, environmentName, location))
+var tags = { 'azd-env-name': environmentName }
+
+// Organize resources in a resource group
+resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: !empty(resourceGroupName) ? resourceGroupName : '${abbrs.resourcesResourceGroups}${environmentName}'
+  location: location
+  tags: tags
+}
+
+// Create an App Service Plan to group applications under the same payment plan and SKU
+module appServicePlan './app/appserviceplan.bicep' = {
+  name: 'appserviceplan'
+  scope: rg
+  params: {
+    name: !empty(appServicePlanName) ? appServicePlanName : '${abbrs.webServerFarms}${resourceToken}'
+    location: location
+    tags: tags
+    sku: {
+      name: 'B1'
+    }
+  }
+}
+
+// Monitor application with Azure Monitor
+module monitoring './monitor/monitoring.bicep' = {
+  name: 'monitoring'
+  scope: rg
+  params: {
+    location: location
+    tags: tags
+    logAnalyticsName: !empty(logAnalyticsName) ? logAnalyticsName : '${abbrs.operationalInsightsWorkspaces}${resourceToken}'
+    applicationInsightsName: !empty(applicationInsightsName) ? applicationInsightsName : '${abbrs.insightsComponents}${resourceToken}'
+    applicationInsightsDashboardName: !empty(applicationInsightsDashboardName) ? applicationInsightsDashboardName : '${abbrs.portalDashboards}${resourceToken}'
+  }
+}
+
+// Store secrets in a keyvault
+module keyVault './security/keyvault.bicep' = {
+  name: 'keyvault'
+  scope: rg
+  params: {
+    name: !empty(keyVaultName) ? keyVaultName : '${abbrs.keyVaultVaults}${resourceToken}'
+    location: location
+    tags: tags
+    principalId: principalId
+  }
+}
+
+// The application database
+module mysql './database/mysql.bicep' = {
+  name: 'mysql'
+  scope: rg
+  dependsOn: [ keyVault ]
+  params: {
+    name: !empty(mysqlServerName) ? mysqlServerName : '${abbrs.dBforMySQLServers}${resourceToken}'
+    databaseName: mysqlDatabaseName
+    location: location
+    tags: tags
+    mysqlAdminName: mysqlAdminName
+    mysqlAdminPassword: mysqlAdminPassword
+    keyVaultName: keyVault.outputs.name
+  }
+}
+
+// The application backend
+module app './app/app.bicep' = {
+  name: 'app'
+  scope: rg
+  dependsOn: [ appServicePlan, monitoring, mysql ]
```

Comment:
You shouldn't need any dependsOn if the other resource is referenced by properties.  Can you scrub all bicep and remove dependsOn?

Reply:
sure, I will remove them.